### PR TITLE
4213 improve views coverage

### PR DESCRIPTION
**Pull request by anagradova**
## PR Checklist: Submitter

**Link to Issue**
Link to main [issue](https://github.com/GSA-TTS/FAC/issues/4213).

**Note**
Reformatted file with black, isort, and ran a flake8 test. 

**Added Tests:** 
EditSubmissionViewTests.setUp
EditSubmissionViewTests.test_redirect_not_logged_in
SubmissionViewTests.setUp
SubmissionViewTests.test_get_renders_template
SubmissionViewTests.def test_get_permission_denied_if_no_sac
SubmissionViewTests.def test_get_access_denied_for_unauthorized_user
SubmissionViewTests.test_post_valid_submission
SubmissionViewTests.test_post_validation_errors
SubmissionViewTests.test_post_transaction_error
SubmissionViewTests.test_post_permission_denied_if_no_sac
SubmissionGetTest.testValidSubmission
SubmissionStatusTests.test_get_login_required
SubmissionStatusTests.test_get_bad_report_id_returns_403
EditSubmissionTest.setUp
EditSubmissionTest.test_redirects_to_singleauditchecklist
AuditorCertificationStep1ViewTests.setUp
AuditorCertificationStep1ViewTests.test_get_redirects_if_status_not_ready_for_certification
AuditorCertificationStep1ViewTests.test_get_renders_template_if_valid_state
AuditorCertificationStep1ViewTests.test_get_permission_denied_if_sac_not_found
AuditorCertificationStep1ViewTests.test_post_redirects_if_status_not_ready_for_certification
AuditorCertificationStep1ViewTests.test_post_valid_form
AuditorCertificationStep1ViewTests.test_post_invalid_form
AuditorCertificationStep1ViewTests.test_post_permission_denied_if_sac_not_found
AuditeeCertificationStep2ViewTests.setUp
AuditeeCertificationStep2ViewTests.test_get_redirects_if_no_step_1
AuditeeCertificationStep2ViewTests.test_get_renders_template_if_valid_session
AuditeeCertificationStep2ViewTests.test_get_redirects_if_not_auditor_certified
AuditeeCertificationStep2ViewTests.test_redirects_if_no_step_1_session
AuditeeCertificationStep2ViewTests.test_post_valid_form
AuditeeCertificationStep2ViewTests.test_post_invalid_form
AuditeeCertificationStep2ViewTests.test_post_redirects_if_status_not_auditor_certified
AuditeeCertificationStep2ViewTests.test_single_audit_checklist_does_not_exist_exception
CrossValidationViewTests.setUp
CrossValidationViewTests.test_get_view_renders_correct_template
CrossValidationViewTests.test_get_view_permission_denied
CrossValidationViewTests.test_post_view_renders_results_template
CrossValidationViewTests.test_post_view_permission_denied

**Test Command**
To test the file only, use **manage.py test audit.test_views**

**Last Merge**
Main was merged to branch on 12/02/24

## PR Checklist: Reviewer

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

## Pre Merge Checklist: Merger

- [ ]   Ensure that prior to approving, the terraform plan is what we expect it to be. `-/+ resource "null_resource" "cors_header" ` should be destroying and recreating its self and ` ~ resource "cloudfoundry_app" "clamav_api" ` might be updating its `sha256` for the `fac-file-scanner` and `fac-av-${ENV}` by default.
- [ ]   Ensure that the branch is up to date with `main`.
- [ ]   Ensure that a terraform plan has been recently generated for the pull request.

### CODE REVIEW

File: backend/audit/test_views.py [hunk 1]
```diff
@@ -1656,3 +1813,316 @@ def test_valid_file_upload_for_notes_to_sefa(self, mock_scan_file):
             submission_events[event_count - 1].event,
             SubmissionEvent.EventType.NOTES_TO_SEFA_UPDATED,
         )
+
+
+class EditSubmissionTest(TestCase):
```
Comment (phildominguez-gsa) [inline backend/audit/test_views.py:1818]: Should these be in the same class as `EditSubmissionViewTests` above?

File: backend/audit/test_views.py [hunk 2]
```diff
@@ -264,16 +265,128 @@ def test_user_with_no_submissions_should_return_no_data(self):
...
+        self.report_id = "TEST_REPORT_ID"
+
     def test_redirect_if_not_logged_in(self):
         result = self.client.get(reverse(EDIT_PATH, args=["SOME_REPORT_ID"]))
         self.assertAlmostEqual(result.status_code, 302)
 
+    def test_redirect_not_logged_in(self):
+        url = reverse(self.url_name, args=[self.report_id])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 302)
```
[the remark this answers is not in the document]
Comment (anagradova) [reply backend/audit/test_views.py]: Yes, they are essentially testing the same thing. I removed one.

File: backend/audit/test_views.py [hunk 3]
```diff
@@ -264,16 +265,128 @@ def test_user_with_no_submissions_should_return_no_data(self):
...
+        self.url = reverse("audit:Submission", kwargs={"report_id": self.sac.report_id})
+        self.client.force_login(self.user)
+        baker.make(
+            "audit.Access",
+            sac=self.sac,
+            user=self.user,
+            role="certifying_auditee_contact",
+        )
+
+    def test_get_renders_template(self):
```
[the remark this answers is not in the document]
Comment (anagradova) [reply backend/audit/test_views.py:334]: docstrings added

File: backend/audit/test_views.py (hunk 1 above)
Comment (anagradova) [reply backend/audit/test_views.py:1861]: Moved and merged into the other class. Originally they were testing one with a logged in session and one without, but I believe I know handled both cases.

**Review by phildominguez-gsa — APPROVED** [review]

**Review by jperson1 — COMMENTED** [review]
These changes look great, I'm just being picky

File: backend/audit/test_views.py [hunk 4]
```diff
@@ -264,16 +274,156 @@ def test_user_with_no_submissions_should_return_no_data(self):
...
+
+        mock_validate.return_value = []
+        mock_general_get.return_value = True
+
+        response = self.client.post(self.url)
+
+        self.assertEqual(response.status_code, 302)
+        self.assertRedirects(response, reverse("audit:MySubmissions"))
+
+        print("this response", response)
```
[the remark this answers is not in the document]
Comment (anagradova) [reply backend/audit/test_views.py]: Done!

File: backend/audit/test_views.py [hunk 5]
```diff
@@ -1656,3 +1856,343 @@ def test_valid_file_upload_for_notes_to_sefa(self, mock_scan_file):
...
+            "is_auditee_responsible": True,
+            "has_used_auditors_report": True,
+            "has_no_auditee_procedures": True,
+            "is_FAC_releasable": True,
+            "is_accurate_and_complete": True,
+        }
+
+        response = self.client.post(self.url, form_data)
+
+        # print(response.content)
```
[the remark this answers is not in the document]
Comment (anagradova) [reply backend/audit/test_views.py]: Done!

**Review by jperson1 — APPROVED** [review]

**Review by jperson1 — APPROVED** [review]

**Review by phildominguez-gsa — APPROVED** [review]

**Review by jperson1 — APPROVED** [review]

**Comment by anagradova** [comment]
Changes added and merged in PR #4546

**Pull request closed without merging**